test/core/derivatives/derivatives_test.py

```diff
@@ -62,14 +59,51 @@
 SUBSAMPLING_IDS = [f"subsampling={s}".replace(" ", "") for s in SUBSAMPLINGS]
 
 
+@mark.parametrize("subsampling", SUBSAMPLINGS, ids=SUBSAMPLING_IDS)
+@mark.parametrize("sum_batch", [True, False], ids=["sum_batch=True", "sum_batch=False"])
+def test_param_mjp(
+    problem: DerivativesTestProblem,
+    sum_batch: bool,
+    subsampling: List[int] or None,
+    request,
+) -> None:
+    """Test all parameter derivatives.
+
+    Args:
+        problem: test problem
+        sum_batch: whether to sum along batch axis
+        subsampling: subsampling indices
+        request: problem request
+    """
+    skip_subsampling_conflict(problem, subsampling)
+    test_save_memory: bool = "Conv" in request.node.callspec.id
+
+    for param_str, _ in problem.module.named_parameters():
+        print(f"testing derivative wrt {param_str}")
+
+        V = 3
+        mat = rand_mat_like_output(V, problem, subsampling=subsampling)
+
+        for save_memory in [True, False] if test_save_memory else [None]:
+            with weight_jac_t_save_memory(
+                save_memory=save_memory
+            ) if test_save_memory else nullcontext():
+                if test_save_memory:
+                    print(f"testing with save_memory={save_memory}")
+                backpack_res = BackpackDerivatives(problem).param_mjp(
+                    param_str, mat, sum_batch, subsampling=subsampling
+                )
+        autograd_res = AutogradDerivatives(problem).param_mjp(
+            param_str, mat, sum_batch, subsampling=subsampling
+        )
+
+        check_sizes_and_values(autograd_res, backpack_res)
```

Comment:
This does not seem to perform the final check for all values of `save_memory` and must be indented

Reply:
This structure is still sub-optimal because there could be side effects from previous `param_str` or `save_memory` runs

Reply:
I cleaned it up a little bit:
- indented check
- order
  1. test settings
  2. loops and print-out
  3. variables and computations
  4. check and delete